woocommerce/woocommerce-ios
# [Custom Amounts M4] Add percentage to custom amounts

Closes: #11139 


## Description

After receiving some user feedback, we restore here the option of adding a custom amount (fee) via an order total percentage.

Unlike before, we consider fees part of the total when calculating percentages. The reason behind this is twofold, we want to make them real elements of an order and thus a user might want to include them when calculating percentages, and want to simplify the implementation by just taking the order total.

Apart from that, we remove with this PR the code relevant to passing the total for calculating the percentage in the previous logic.

## Testing instructions


1. Go to Orders
2. Tap on + to create a new order
3. Add a product
4. Tap on Add Custom Amount
5. See that we can enter a percentage of the order total. See that we enter a percentage, and the amount is updated.
6. When tapping in Done, the custom amount is created with the percentage based amount. The event is tracked:
`🔵 Tracked add_custom_amount_percentage_added, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 214354650, AnyHashable("plan"): "business-bundle", AnyHashable("site_url"): "https://americanwootester.wpcomstaging.com", AnyHashable("was_ecommerce_trial"): false]`
7. See that you can edit the custom amount. As explained before, the order total contains now the added custom amount. If you want to re-calculate the percentage of the order total without that custom amount, the user will have to delete the custom amount first.

- See also that when the order is empty and you tap on Add Custom Amount, there's no percentage input field, as there's no order total yet.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/1864060/6ed2ea39-75ae-4ea9-b19c-1e0ac65d4455




---
